src/redux/sagas/motions/createEditDomainMotion.ts

```diff
@@ -35,10 +33,172 @@ import {
   uploadAnnotation,
   initiateTransaction,
   createActionMetadataInDB,
+  getPermissionProofsLocal,
+  getChildIndexLocal,
 } from '../utils/index.ts';
 
 // import { ipfsUpload } from '../ipfs';
 
+type CreateTransactionPayload = {
+  isMultiSig: boolean;
+  colonyAddress: Address;
+  batchKey: string;
+  metaId: string;
+  index: number;
+  colonyDomains: Domain[];
+  isCreateDomain: boolean;
+  domainNativeId: number;
+  action: string;
+  colonyRoles: ColonyRoleFragment[];
+  domain: Domain | undefined;
+  parentDomainId: number;
+};
+
+function* getCreateTransactionPayload({
+  isMultiSig,
+  colonyAddress,
+  batchKey,
+  metaId,
+  index,
+  colonyDomains,
+  isCreateDomain,
+  domainNativeId,
+  action,
+  colonyRoles,
+  domain,
+  parentDomainId,
+}: CreateTransactionPayload) {
+  const transactionParams = {
+    context: isMultiSig
+      ? ClientType.MultisigPermissionsClient
+      : ClientType.VotingReputationClient,
+    methodName: TRANSACTION_METHODS.CreateMotion,
+    identifier: colonyAddress,
+    group: {
+      key: batchKey,
+      id: metaId,
+      index,
+    },
+    params: [] as MethodParams,
+    ready: false,
+  };
+
+  const colonyManager = yield getColonyManager();
+
+  const colonyClient = yield colonyManager.getClient(
+    ClientType.ColonyClient,
+    colonyAddress,
+  );
+
+  // eslint-disable-next-line no-warning-comments
+  // TODO: Use Sam's upcoming list of required permissions as the source of truth
+  // src/constants/actions.ts: PERMISSIONS_NEEDED_FOR_ACTION
+  const requiredRoles = [ColonyRole.Architecture];
+
+  const userAddress = yield colonyClient.signer.getAddress();
+
+  if (isMultiSig) {
+    const [permissionDomainId, childSkillIndex] = yield call(
+      getPermissionProofsLocal,
+      {
+        networkClient: colonyClient.networkClient,
+        colonyRoles,
+        colonyDomains,
+        requiredDomainId: parentDomainId, // You need Architecture permissions in the parent domain to edit a subdomain
+        requiredColonyRole: requiredRoles,
+        permissionAddress: userAddress,
+        isMultiSig,
+      },
+    );
+
+    const encodedAction = colonyClient.interface.encodeFunctionData(
+      action,
+      [permissionDomainId, childSkillIndex, domainNativeId, '.'], // domainMetadataIpfsHash
+    );
+
+    transactionParams.params = [
+      Id.RootDomain,
```

Comment:
This is restricting it to root domain rather than creating it in the child domain.